conftest.py

```diff
@@ -51,3 +55,47 @@ def pytest_itemcollected(item):
 
     if isinstance(item, pytest.DoctestItem):
         item.add_marker(doctest_marker)
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    """Hook to register custom CLI options and config values"""
+    parser.addoption(
+        "--strict-warnings",
+        action="store_true",
+        default=False,
+        help="Turn most warnings into errors",
```

Comment:
This seems a little vague, maybe something like "Applies the `filterwarnings_when_strict` option from the `pyproject.toml` to a test run."  I think it would be good to make clear how this flag works.